src/views/connect/components/dynamodb-connect-dialog.vue

```diff
@@ -0,0 +1,230 @@
+  { label: 'Asia Pacific (Mumbai)', value: 'ap-south-1' },
+  { label: 'Asia Pacific (Seoul)', value: 'ap-northeast-2' },
+  { label: 'Asia Pacific (Singapore)', value: 'ap-southeast-1' },
+  { label: 'Asia Pacific (Sydney)', value: 'ap-southeast-2' },
+  { label: 'Asia Pacific (Tokyo)', value: 'ap-northeast-1' },
+  { label: 'Canada (Central)', value: 'ca-central-1' },
+  { label: 'Europe (Frankfurt)', value: 'eu-central-1' },
+  { label: 'Europe (Ireland)', value: 'eu-west-1' },
+];
+
+const defaultFormData: DynamoDBConnection = {
+  name: '',
+  type: DatabaseType.DYNAMODB,
+  region: '',
+  accessKeyId: '',
+  secretAccessKey: '',
+};
+
+const formData = ref<DynamoDBConnection>({ ...defaultFormData });
+
+const formRules = reactive({
+  name: [
+    {
+      required: true,
+      message: lang.t('connection.formValidation.nameRequired'),
+      trigger: ['input', 'blur'],
+    },
+  ],
+  region: [
+    {
+      required: true,
+      message: lang.t('connection.formValidation.regionRequired'),
+      trigger: ['input', 'blur'],
+    },
+  ],
+  accessKeyId: [
+    {
+      required: true,
+      message: lang.t('connection.formValidation.accessKeyIdRequired'),
+      trigger: ['input', 'blur'],
+    },
+  ],
+  secretAccessKey: [
+    {
+      required: true,
+      message: lang.t('connection.formValidation.secretAccessKeyRequired'),
+      trigger: ['input', 'blur'],
+    },
+  ],
+});
+
+const message = useMessage();
+
+const cleanUp = () => {
+  formData.value = { ...defaultFormData };
+  modalTitle.value = lang.t('connection.add');
+};
+
+const showMedal = (con: DynamoDBConnection | null) => {
+  cleanUp();
+  showModal.value = true;
+  if (con) {
+    formData.value = { ...con };
+    modalTitle.value = lang.t('connection.edit');
+  }
+};
+
+const closeModal = () => {
+  showModal.value = false;
+  cleanUp();
+};
+
+const validationPassed = computed(() => {
+  return (
+    formData.value.name &&
+    formData.value.region &&
+    formData.value.accessKeyId &&
+    formData.value.secretAccessKey
+  );
+});
+
+const testConnect = async () => {
+  try {
+    testLoading.value = true;
+    await connectionStore.testDynamoDBConnection(formData.value);
+    message.success(lang.t('connection.testSuccess'));
+  } catch (error) {
+    if (error instanceof Error) {
+      message.error(error.message);
+    } else {
+      message.error(lang.t('connection.unknownError'));
+    }
+  } finally {
+    testLoading.value = false;
+  }
+};
+
+const saveConnect = async () => {
+  try {
+    saveLoading.value = true;
+    await connectionStore.saveConnection(formData.value);
+    message.success(lang.t('connection.saveSuccess'));
+    closeModal();
+  } catch (error) {
+    if (!(error instanceof Error && error.message.includes('__TAURI_IPC__'))) {
+      if (error instanceof Error) {
+        message.error(error.message);
+      } else {
+        message.error(lang.t('connection.unknownError'));
+      }
```

Comment:
it looks worth encapsulating the error inside saveConnection, so On UI page it only needs to care about showing the message ?